sz-poc-offsite-2025/evm/prover/src/main.rs

```diff
@@ -0,0 +1,154 @@
+use clap::Parser;
+use std::{path::PathBuf, process::Command, thread, time::Duration};
+
+#[derive(Parser, Debug)]
+#[clap(author, version, about = "Ethereum Proof Generation Tool")]
+struct Args {
+    #[clap(long, default_value = "http://localhost:8546")]
+    rpc: String,
+
+    #[clap(long, default_value = "5")]
+    start_block: u64,
+
+    #[clap(long, default_value = "10")]
+    batch_size: u64,
+
+    #[clap(long, default_value = "5")]
+    interval: u64,
+
+    #[clap(long)]
+    zeth_binary_dir: Option<String>,
+}
+
+fn run_ethereum_prove(
+    rpc: &str,
+    block_number: u64,
+    batch_size: u64,
+    zeth_binary_dir: Option<String>,
+) -> Result<(), String> {
+    println!(
+        "Running Ethereum prove for blocks {}-{}",
+        block_number,
+        block_number + batch_size - 1
+    );
+
+    let binary_path = if let Some(dir) = &zeth_binary_dir {
+        let mut path = PathBuf::from(dir);
+        path.push("zeth-ethereum");
+        path
+    } else {
+        return Err("No binary directory provided".to_string());
+    };
+
+    if !binary_path.exists() {
+        return Err(format!("Binary not found at: {}", binary_path.display()));
+    }
+
+    let output = Command::new(&binary_path)
+        .env("RUST_LOG", "info")
+        .args([
+            "prove",
+            &format!("--rpc={}", rpc),
+            &format!("--block-number={}", block_number),
+            &format!("--block-count={}", batch_size),
+        ])
+        .output()
+        .map_err(|e| format!("Failed to execute zeth-ethereum prove: {}", e))?;
+
+    if !output.status.success() {
+        return Err(format!(
+            "ethereum prove command failed with status: {}\nStderr: {}",
+            output.status,
+            String::from_utf8_lossy(&output.stderr)
+        ));
+    }
+
+    println!("Successfully processed batch");
+    Ok(())
+}
+
+fn get_latest_block(rpc: &str) -> Result<u64, String> {
+    println!("Checking latest block height...");
+
+    let output = Command::new("curl")
```

Comment:
we could use reqwest so that sending and receiving json values is quite a bit easier